akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -257,3 +233,89 @@ private[akka] final class HeadOptionStage[T] extends GraphStageWithMaterializedV
     }, p.future)
   }
 }
+
+/**
+ * INTERNAL API
+ */
+private[akka] class QueueSink[T]() extends GraphStageWithMaterializedValue[SinkShape[T], SinkQueue[T]] {
+  trait RequestElementCallback[E] {
+    val requestElement = new AtomicReference[Either[List[E], E ⇒ Unit]](Left(Nil))
```

Comment:
for these cases I’d actually just use AnyRef and store the different things directly—we need pattern matching to find out the difference in any case